gssapi/raw/sec_contexts.pyx

```diff
@@ -232,7 +232,9 @@ flags=None, lifetime=None, channel_bindings=None, input_token=None)
 
     cdef OID output_mech_type = OID()
     if maj_stat == GSS_S_COMPLETE or maj_stat == GSS_S_CONTINUE_NEEDED:
-        output_mech_type.raw_oid = actual_mech_type[0]
+        if actual_mech_type:
```

Comment:
you might want to make this `is not GSS_C_NO_OID` or `!= GSS_C_NO_OID` (your code works fine because `GSS_C_NO_OID` is actually `NULL`, but it's a bit clearer)